src/middleware.rs

```diff
@@ -367,3 +432,9 @@ pub struct MainPodInputs<'a> {
 pub trait PodProver {
     fn prove(&mut self, params: &Params, inputs: MainPodInputs) -> Result<Box<dyn MainPod>>;
 }
+
+pub trait ToFields {
+    /// returns Vec<F> representation of the type, and a usize indicating how many field elements
+    /// does the vector contain
+    fn to_fields(self) -> (Vec<F>, usize);
```

Comment:
@arnaucube what's the reason to include the length as a value when the Vec already contains the same value with `Vec::len`?